### ISSUE TITLE
deploy-sourcegraph: Updating image tags is error-prone and tedious

### ISSUE THREAD

**Issue by beyang** [sourcegraph/sourcegraph#9725]
From @slimsag (migrated from https://github.com/sourcegraph/deploy-sourcegraph/issues/287):

When cutting a new release, the image tags in the YAML here must be updated which is error-prone and tedious work.

In the best case, Renovate creates the appropriate PR and you just merge + cherry pick it onto e.g. the `3.5` branch and tag the release. However:

1. Renovate does not always include the relevant tags in the update if it doesn't have time to discover them. When I merged https://github.com/sourcegraph/deploy-sourcegraph/pull/278 all images were released and on Docker Hub, but it did not update the `frontend` tag and I missed this which caused the 3.5.1 release to incorrectly not bump the frontend version. The only way I could have caught this was by clicking `rebase` and waiting for it to rebase OR by manually comparing the list of updated images against all that should be updated.

2. When the versions are incorrect, as was the case after I merged https://github.com/sourcegraph/deploy-sourcegraph/pull/278 -- Renovate created this PR which upgraded the version from 3.5.0 -> 3.5.1 *instead of* to 3.5.2 as I expected. https://github.com/sourcegraph/deploy-sourcegraph/pull/282

The TL;DR here seems to be that I should not trust the PRs renovate creates and need to actually review them -- despite the fact that they are machine generated.

Proposal: Can we just have a `VERSION=3.5.2 ./update-image-tags.sh` script instead? This would be much faster than Renovate and would give me more confidence in the change.

--------------------

This just happened again. Luckily not an actual issue this time around, though. https://sourcegraph.slack.com/archives/C0EPTDE9L/p1566521183091600?thread_ts=1566501269.087000&cid=C0EPTDE9L

**Comment by slimsag** [comment]
Other cases where Renovate updates of these image tags may fail us currently:

- When we have an "insiders" image tag in the form of `60605_2020-04-09_34ab38e`, Renovate has no way of knowing "released" `v3.14.2` version is newer so the image tags must be updated manually: https://sourcegraph.slack.com/archives/CJX299FGE/p1586386442263100
- If Renovate did try to update "60605_2020-04-09_34ab38e" to the latest release (v3.14.2), it would've introduced bugs in `master` (i.e. Renovate would need to understand the relevant ordering of `v3.14.2` and non-semver `60605_2020-04-09_34ab38e` to handle this case): https://github.com/sourcegraph/deploy-sourcegraph/pull/614
- Renovate will need to be configured to run against each "release" branch we have, but NOT merge e.g. v3.15.0 into the 3.14 branch when it is released. https://github.com/sourcegraph/deploy-sourcegraph/blob/5de3b60fd98de70b85ba5c7931f43546e59e7ea8/renovate.json#L27

Overall I just think Renovate isn't the right tool for the job here -- it's great for more conventional dependency updates but for our release pipeline a script would do us better.

**Comment by slimsag** [comment]
Update: I have a script that does the job here well, it just is not compatible with `zoekt` images primarily because we have released a `2018.0.1` tag which is semver > what is actually latest currently -- we'll need to fix our image tag versioning mess before we can use it nicely.

**Comment by beyang** [comment]
@slimsag is there more work to be done here, or can we close?

**Comment by slimsag** [comment]
Closing in favor of https://github.com/sourcegraph/sourcegraph/issues/10011
